core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -73,47 +138,68 @@ object JsonAST {
       case _ => Nil
     }
 
-    /** XPath-like expression to query JSON fields by name. Returns all matching fields.
-      * <p>
-      * Example:<pre>
-      * json \\ "name"
-      * </pre>
-      */
+    /**
+     * Find all children of a `JObject` with the matching name, returning an empty `JObject` is no matches are found.
+     *
+     * Example:
+     *
+     * {{{
+     * json \\ "name"
+     * }}}
+     */
     def \\(nameToFind: String): JValue = {
       def find(json: JValue): List[JField] = json match {
-        case JObject(l) => l.foldLeft(List[JField]()) { 
-          case (a, JField(name, value)) => 
-            if (name == nameToFind) {
-              a ::: List(JField(name, value)) ::: find(value)
-            } else {
-              a ::: find(value)
-            }
-        }
-        case JArray(l) => l.foldLeft(List[JField]())((a, json) => a ::: find(json))
-        case _ => Nil
+        case JObject(fields) =>
+          fields.foldLeft(List[JField]()) {
+            case (matchingFields, JField(name, value)) =>
+              matchingFields :::
+              List(JField(name, value)).filter(_.name == nameToFind) :::
```

Comment:
I guess I've always looked at it as dependency between lines? So the fact that they're one statement to the compiler is immaterial to me if the individual lines could be removed and it would still have a valid effect.